cli/azd/cmd/container.go

```diff
@@ -28,21 +28,23 @@ import (
 
 // Registers a singleton action for the specified action name
 // This finds the action for a named instance and casts it to the correct type for injection
-func registerAction[T any](container *ioc.NestedContainer, actionName string) {
-	container.RegisterSingleton(func() (T, error) {
-		var zero T
-		var action actions.Action
-		err := container.ResolveNamed(actionName, &action)
-		if err != nil {
-			return zero, err
-		}
-
-		instance, ok := action.(T)
-		if !ok {
-			return zero, fmt.Errorf("failed converting action to '%T'", zero)
+func registerAction[T actions.Action](container *ioc.NestedContainer, actionName string) {
+	container.RegisterSingleton(func() actions.ActionInitializer[T] {
+		return func() T {
+			var zero T
+			var action actions.Action
+			err := container.ResolveNamed(actionName, &action)
+			if err != nil {
+				panic(fmt.Errorf("failed resolving action '%s', %w", actionName, err))
```

Comment:
Thoughts on whether we should panic here or continue just returning an error like before?

Reply:
I think we still want the error formatting thing (remove golobby error messaging) to take effect, yes? Returning an `error` might be the way to do that.